### PR TITLE
Rename current Cluster types to LegacyCluster

### PR DESCRIPTION
**Pull request by tuommaki**
As a preparation for introducing controller to reconcile Cluster CR, rename
current Cluster types to LegacyCluster.

Plan here is to eventually reconcile only on Cluster CR and have configuration
to extract common part of provider specific extension of it. This way we can
simplify the code and make it a bit more straightforward.

In the first phase the Cluster CR reconciliation is used to map information
from Cluster CR to AWSClusterConfig and drive cluster-operator logic that way.

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -199,20 +199,20 @@ func New(config Config) (*Service, error) {
...
 
-		azureClusterController, err = azure.NewCluster(c)
+		azureClusterController, err = azure.NewLegacyCluster(c)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 	}
 
-	var kvmClusterController *kvm.Cluster
+	var kvmClusterController *kvm.LegacyCluster
```
Comment (xh3b4sd) [inline service/service.go]: I think the variable names should be aligned as well as we are going to name the other controllers somehow. What was your idea there?

Comment (tuommaki) [reply service/service.go]: Didn't outright have an idea. There should be only one cluster controller later, but yeah, it makes sense to align these. One minute, please. :slightly_smiling_face:

**Comment by tuommaki** [comment]
> Are we usually put legacy as prefix in this case?

I think this depends a bit case by case. For me difference between `kvmLegacyCluster` and `legacyKVMCluster` is that former suggests there is this `LegacyCluster` and different provider specific flavours while the latter one suggests that there's non-legacy `KVMCluster` somewhere to be found while that won't be the case with current understanding.

File: service/controller/aws/cluster.go [hunk 2]
```diff
@@ -47,12 +47,12 @@ type ClusterConfig struct {
 	ResourceNamespace  string
 }
 
-type Cluster struct {
+type LegacyCluster struct {
```
Comment (kopiczko) [inline service/controller/aws/cluster.go]: I think file should be `legacy_cluster.go`?

Comment (tuommaki) [reply service/controller/aws/cluster.go]: Fine. Done.

**Pull request merged**